### PR TITLE
refactor: replace specific requirePermissionDecorator with more gener…

### PR DESCRIPTION
**Pull request by markkopaas**
…ic requirePredicate

### CODE REVIEW

File: src/pages/concept-list-page/concept-list-page.ts [hunk 1]
```diff
@@ -3,18 +3,22 @@ import { compose, withProps } from 'recompose';
 
 import { ConceptListPagePure } from './concept-list-page-pure';
 import { conceptListResolver } from './concept-list-resolver';
-import { requirePermissionDecorator } from '../../auth/require-permission-decorator';
-import { PERMISSION_READ } from '../../auth/auth-service';
+import { hasPermissionForResource, logout, PERMISSION_READ } from '../../auth/auth-service';
```
Comment (azaxarov) [inline src/pages/concept-list-page/concept-list-page.ts]: Convert PERMISSION_READ to Permission enum with READ, WRITE and ADMIN properties

File: src/pages/concept-list-page/concept-list-page.ts [hunk 2]
```diff
@@ -3,18 +3,22 @@ import { compose, withProps } from 'recompose';
...
-    permission: PERMISSION_READ
-  }),
+  requirePredicate(
+    ({ catalogId }) =>
+      hasPermissionForResource({
+        resource: 'organization',
+        resourceId: catalogId,
+        permission: PERMISSION_READ
+      }),
+    () => logout()
```
Comment (azaxarov) [inline src/pages/concept-list-page/concept-list-page.ts]: How about just sending logout as a reference?

File: src/pages/concept-list-page/concept-list-page.ts [hunk 3]
```diff
@@ -3,18 +3,22 @@ import { compose, withProps } from 'recompose';
...
   mapRouteParams,
-  requirePermissionDecorator({
-    resource: 'organization',
-    resourceIdPropName: 'catalogId',
-    permission: PERMISSION_READ
-  }),
+  requirePredicate(
+    ({ catalogId }) =>
+      hasPermissionForResource({
+        resource: 'organization',
```
Comment (azaxarov) [inline src/pages/concept-list-page/concept-list-page.ts]: Also enum

File: src/pages/concept-list-page/concept-list-page.ts (hunk 2 above)
Comment (markkopaas) [reply src/pages/concept-list-page/concept-list-page.ts]: this is temporary refactoring, better implementation in next commit

File: src/pages/concept-list-page/concept-list-page.ts (hunk 1 above)
Comment (markkopaas) [reply src/pages/concept-list-page/concept-list-page.ts]: this is temporary refactoring, better implementation in next commit

File: src/pages/concept-list-page/concept-list-page.ts (hunk 3 above)
Comment (markkopaas) [reply src/pages/concept-list-page/concept-list-page.ts]: will be replaced by function

**Pull request closed without merging**